src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -5163,6 +5163,86 @@ void C2_MacroAssembler::vector_castD2X_evex(BasicType to_elem_bt, XMMRegister ds
   }
 }
 
+void C2_MacroAssembler::vector_castF2X_avx10(BasicType to_elem_bt, XMMRegister dst, XMMRegister src, int vec_enc) {
+  switch(to_elem_bt) {
+    case T_LONG:
+      evcvttps2qqs(dst, src, vec_enc);
+      break;
+    case T_INT:
+      evcvttps2dqs(dst, src, vec_enc);
+      break;
+    case T_SHORT:
+      evcvttps2dqs(dst, src, vec_enc);
+      evpmovdw(dst, dst, vec_enc);
+      break;
+    case T_BYTE:
+      evcvttps2dqs(dst, src, vec_enc);
+      evpmovdb(dst, dst, vec_enc);
+      break;
+    default: assert(false, "%s", type2name(to_elem_bt));
```

Comment:
Perhaps you could provide a richer assert message like this:
```suggestion
    default: assert(false, "unexpexted basic type for vector castF2X AVX10: %s", type2name(to_elem_bt));
```

Reply:
Sure, I added more verbose messages for the other vector casts as well.